Dodge.js

```diff
@@ -0,0 +1,179 @@
+import { StatusBar } from 'expo-status-bar';
+import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
+import AxisPad from 'react-native-axis-pad';
+////this looks interesting for the attacker placing/throwing projectiles:
+//https://reactnative.dev/docs/panresponder 
+//https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
+var ents = {}
+
+const generateUniqueId = require('generate-unique-id');
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+
+
+//adds an entity given its type, if it's a Grenade set a timer for explosion
+function AddEntity(entity){
+let id = generateUniqueId()
+ents[id] = new entity(id)
+
+if(entity == Grenade){
+  setTimeout(() => { Explode(id) }, 2000)
+}
+}
+
+//deletes a single entity given that entities unique id
+function DeleteEntity(id){
+delete(ents[id])
+}
+
+//blows up the grenade
+function Explode(id){
+  //spawns shrapnel traveling from where the grenade blew up
+  InitialShrapnel(ents[id].x,ents[id].y)
+  //delete the original projectile
+  DeleteEntity(id)
+}
+
+//runs 60 times a second updating each entitiy
+function Gameloop(entities, { touches }, )  {
+  let toDelete = []
+  for (let id in entities) {
+      entities[id].update()
+  }
+  
+return entities
+}
+
+//creates a square component
+function CreateBrick(props) {
```

Comment:
You could move the CreateX functions to the Entities file, since that's where they're now used.